File: airbyte-db/db-lib/src/test/java/io/airbyte/db/check/impl/ConfigsDatabaseAvailabilityCheckTest.java
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.db.check.impl;

import static org.mockito.Mockito.any;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import io.airbyte.db.check.ConfigsDatabaseAvailabilityCheck;
import io.airbyte.db.check.DatabaseCheckException;
import org.jooq.DSLContext;
import org.jooq.Select;
import org.jooq.exception.DataAccessException;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Test suite for the {@link ConfigsDatabaseAvailabilityCheck} class.
 */
class ConfigsDatabaseAvailabilityCheckTest extends CommonDatabaseCheckTest {

  @Test
  void checkDatabaseAvailability() {
    final var check = new ConfigsDatabaseAvailabilityCheck(dslContext, TIMEOUT_MS);
    Assertions.assertDoesNotThrow(check::check);
  }

  @Test
  void checkDatabaseAvailabilityTimeout() {
    final DSLContext dslContext = mock(DSLContext.class);
    when(dslContext.fetchExists(any(Select.class))).thenThrow(new DataAccessException("test"));
    final var check = new ConfigsDatabaseAvailabilityCheck(dslContext, TIMEOUT_MS);
    Assertions.assertThrows(DatabaseCheckException.class, check::check);
  }

  @Test
  void checkDatabaseAvailabilityNullDslContext() {
    Assertions.assertThrows(NullPointerException.class, () -> new ConfigsDatabaseAvailabilityCheck(null, TIMEOUT_MS));
  }

}

File: airbyte-db/db-lib/src/test/kotlin/io/airbyte/db/check/ConfigsDatabaseAvailabilityCheckTest.kt
/*
 * Copyright (c) 2020-2025 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.db.check

import io.mockk.every
import io.mockk.mockk
import org.jooq.DSLContext
import org.jooq.Select
import org.jooq.exception.DataAccessException
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertDoesNotThrow
import org.junit.jupiter.api.assertThrows

/**
 * Test suite for the [ConfigsDatabaseAvailabilityCheck] class.
 */
internal class ConfigsDatabaseAvailabilityCheckTest : CommonsDatabaseCheckTest() {
  @Test
  fun checkDatabaseAvailability() {
    val check = ConfigsDatabaseAvailabilityCheck(dslContext, TIMEOUT_MS)
    assertDoesNotThrow { check.check() }
  }

  @Test
  fun checkDatabaseAvailabilityTimeout() {
    val dslContext =
      mockk<DSLContext> {
        every { fetchExists(any<Select<org.jooq.Record>>()) } throws DataAccessException("test")
      }
    val check = ConfigsDatabaseAvailabilityCheck(dslContext, TIMEOUT_MS)
    assertThrows<DatabaseCheckException> { check.check() }
  }
}
